products/securityscanner/api.yaml

```diff
@@ -0,0 +1,157 @@
+    name: Web Security Scanner API
+    url: https://console.cloud.google.com/apis/library/websecurityscanner.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: ScanConfig
+    base_url: projects/{{project}}/scanConfigs
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A ScanConfig resource contains the configurations to launch a scan.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        output: true
+        description: |
+          A server defined name for this index. Format:
+          `projects/{{project}}/scanConfigs/{{server_generated_id}}`
+      - !ruby/object:Api::Type::String
+        name: displayName
+        required: true
+        description: |
+          The user provider display name of the ScanConfig.
+      - !ruby/object:Api::Type::Integer
+        name: maxQps
+        description: |
+          The maximum QPS during scanning. A valid value ranges from 5 to 20 inclusively.
+          If the field is unspecified or its value is set 0, server will default to 15.
+          Other values outside of [5, 20] range will be rejected with INVALID_ARGUMENT error.
+        default_value: 15
+      - !ruby/object:Api::Type::Array
+        name: startingUrls
+        description: |
+          The starting URLs from which the scanner finds site pages.
+        required: true
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::NestedObject
+        name: authentication
+        description: |
+          The authentication configuration.
+          If specified, service will use the authentication configuration during scanning.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: googleAccount
+            description: |
+              Describes authentication configuration that uses a Google account.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: username
+                required: true
+                description: |
+                  The user name of the Google account.
+              - !ruby/object:Api::Type::String
+                name: password
+                input: true
+                required: true
+                description: |
+                  The password of the Google account. The credential is stored encrypted
+                  and not returned in any response nor included in audit logs.
+          - !ruby/object:Api::Type::NestedObject
+            name: customAccount
+            description: |
+              Describes authentication configuration that uses a custom account.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: username
+                required: true
+                description: |
+                  The user name of the custom account.
+              - !ruby/object:Api::Type::String
+                name: password
```

Comment:
I imagine this will need to be marked as sensitive in terraform.yaml (I can't find any docs explaining what this means right now but I might be able to find some later)

Reply:
Still can't find any docs but there's some comments in the code: https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L273-L277